### PR TITLE
[5.1] Store method name, in the mutator cache array

### PR DESCRIPTION
**Pull request by reshadman**
The problem has been described in Issue [#9267](https://github.com/laravel/framework/issues/9276).

### CODE REVIEW

File: src/Illuminate/Database/Eloquent/Model.php [hunk 1]
```diff
@@ -2649,14 +2649,14 @@ public function hasGetMutator($key)
...
-        return $this->{'get'.studly_case($key).'Attribute'}($value);
+        return $this->{array_flip($this->getMutatedAttributes())[$key]}($value);
     }
 
     /**
      * Get the value of an attribute using its mutator for array conversion.
      *
-     * @param  string  $key
-     * @param  mixed   $value
+     * @param  string       $key
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Model.php]: no

**Comment by GrahamCampbell** [comment]
Oh, right. I thought you were talking about caching between requests.

**Comment by reshadman** [comment]
@GrahamCampbell , Well I miss understood, exactly vice versa :D

**Comment by taylorotwell** [comment]
Didn't we try this before and it broke other things?

**Comment by reshadman** [comment]
@taylorotwell well, The unit tests seem working now, but I still thing something like ```php artisan eloquent:cache``` would be great as increasing number of methods in a model impacts the performance of somewhere else.

**Comment by GrahamCampbell** [comment]
> The unit tests seem working now

Yeh. but our tests are pretty poor coverage wise in some areas.

**Comment by taylorotwell** [comment]
Can we get some real numbers regarding the speed-up this provides on say retrieving 100 Eloquent models in a query?

**Comment by taylorotwell** [comment]
Closing pending actual performance benchmarks.

**Pull request closed without merging**